### PR TITLE
feat(inputs): create TextInput component

### PR DESCRIPTION
**Pull request by CoroDaniel**
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-288453

### CODE REVIEW

File: src/components/WebexMeetingGuestAuthentication/WebexMeetingGuestAuthentication.jsx [hunk 1]
```diff
@@ -89,9 +90,8 @@ export default function WebexMeetingGuestAuthentication({
       </div>
       <form className={sc('form-content')}>
         {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
```
Comment (cipak) [inline src/components/WebexMeetingGuestAuthentication/WebexMeetingGuestAuthentication.jsx:92]: is this still needed?

File: src/components/WebexMeetingGuestAuthentication/WebexMeetingGuestAuthentication.jsx [hunk 2]
```diff
@@ -101,6 +101,7 @@ export default function WebexMeetingGuestAuthentication({
           ariaLabel={HINTS.name}
           autoFocus
           tabIndex={101}
+          withClear
```
Comment (cipak) [inline src/components/WebexMeetingGuestAuthentication/WebexMeetingGuestAuthentication.jsx:104]: `withClear` should be a prop of InputField, and the TextInput component should have it hardcoded to `true`

File: src/components/generic/InputField/InputField.jsx [hunk 3]
```diff
@@ -16,45 +16,47 @@ const HINTS = {
  * Input Field component.
```
Comment (cipak) [inline src/components/generic/InputField/InputField.jsx:16]: Generic input field component for building more specific components (text, date, number, etc.). Should not be used directly in forms.

File: src/components/generic/InputField/InputField.jsx [hunk 4]
```diff
@@ -16,45 +16,47 @@ const HINTS = {
...
  *
  * @param {object} props  Data passed to the component
+ * @param {string} [props.ariaLabel]  Hint to be displayed as aria-label
```
Comment (cipak) [inline src/components/generic/InputField/InputField.jsx:19]: just "Aria label", as it's not normally "displayed", but read.

File: src/components/generic/InputField/InputField.jsx [hunk 5]
```diff
@@ -112,8 +114,8 @@ export default function InputField({
             <Icon name={isPwdRevealed ? 'hide-password' : 'show-password'} />
           </Button>
         )}
-        {type !== 'password' && value !== undefined && (
-          <Button type="ghost" className={sc('input-field-right-icon')} size={28} onClick={clearInput}>
+        {withClear && value && (
```
Comment (cipak) [inline src/components/generic/InputField/InputField.jsx:117]: value !== undefined.

Also, the right padding of the input would need to be adjusted to allow space for the icon(s).

I would recommend creating a container for these absolutely positioned icons (clear and eye), putting a ref on it, calling useElementDimensions on it and updating the right padding of the input to the width of this icon container (plus some offset).

File: src/components/generic/InputField/InputField.jsx [hunk 6]
```diff
@@ -13,57 +13,61 @@ const HINTS = {
...
- * @param {string} [props.ariaLabel]  Hint to be displayed as aria-label
+ * @param {number} [props.max]  Maximum value for the input element
+ * @param {number} [props.maxLength]  Maximum number of characters allowed
+ * @param {number} [props.min]  Minimum value for the input element
+ * @param {string} [props.name]  Input name
+ * @param {Function} [props.onChange]  Action to perform on input change
+ * @param {string} [props.pattern]  Specifies a regular expression that the element's value is checked against
+ * @param {string} [props.placeholder]  Input placeholder
  * @param {boolean} [props.required=false]  Flag indicating input required
+ * @param {(boolean|object)} [props.rightIcon]  Icon to be displayed on the right side of the input
```
Comment (cipak) [inline src/components/generic/InputField/InputField.jsx:34]: {boolean|object}

Comment (cipak) [reply src/components/generic/InputField/InputField.jsx:34]: better yet, {false|React.JSX.Element}

Comment (CoroDaniel) [reply src/components/generic/InputField/InputField.jsx:34]: to avoid the warning I put: {boolean|React.JSX.Element}

**Review by cipak — APPROVED** [review]

**Pull request merged**